src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -875,6 +876,15 @@ private void parseCommand() throws SnowflakeSQLException {
           "Failed to parse the locations due to: " + ex.getMessage());
     }
 
+    JsonNode thresholdNode = jsonNode.path("data").path("threshold");
+    int threshold = thresholdNode.asInt();
+    // if value is <= 0, this means an error was made in parsing the threshold or the threshold is
+    // invalid.
+    // Only use the threshold value if it is valid.
+    if (threshold > 0) {
```

Comment:
I thought there was a check for this in GS: https://github.com/snowflakedb/snowflake/blob/01cc79635740ee495d44e1e8e3ae988846e958eb/GlobalServices/src/main/java/com/snowflake/sql/compiler/parsetree/SqlPut.java#L51

Is this just a 2nd safeguard for this, or is this something that we should test for in all of the connectors?

Reply:
I added this check to make the driver backward compatible with GS versions that don't include the threshold in the JSON response. The behavior of the **asInt()** function is to return 0 if the node is null or the value is invalid.

If no threshold is returned, **thresholdNode** will be **null** and **asInt()** will return 0, in which case it's best to go with the default threshold value.